File: Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        d={"(":")","{":"}","[":"]"}
        stack=[]
        for i in s:
            if i in d:
                stack.append(i)
            else:
                if len(stack)==0:
                    return False
                elif d[stack.pop()]!=i:
                    return False
        return True if len(stack)==0 else False
                    

File: Algorithms(Java)/Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        HashMap<Character,Character> map=new HashMap<>();
        map.put('(',')');map.put('{','}');map.put('[',']');
        LinkedList<Character> l=new LinkedList<>();
        for(int i=0;i<s.length();i++){
            if (map.containsKey(s.charAt(i))){
                l.add(s.charAt(i));
            }else{
                if(l.isEmpty()){
                   return false; 
                }else if(map.get(l.removeLast())!=s.charAt(i)){
                    return false;
                }
            }
        }
        return l.size()==0?true:false;
    }
}